staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -545,6 +551,113 @@ message ListPodSandboxResponse {
+}
+
+message ListPodSandboxStatsResponse {
+    // Stats of the pod.
+    repeated PodSandboxStats stats = 1;
+}
+
+// PodSandboxAttributes provides basic information of the container.
+message PodSandboxAttributes {
+    // ID of the pod.
+    string id = 1;
+    // Metadata of the pod.
+    PodSandboxMetadata metadata = 2;
+    // Key-value pairs that may be used to scope and select individual resources.
+    map<string,string> labels = 3;
+    // Unstructured key-value map holding arbitrary metadata.
+    // Annotations MUST NOT be altered by the runtime; the value of this field
+    // MUST be identical to that of the corresponding PodSandboxStatus used to
+    // instantiate the PodSandbox this status represents.
+    map<string,string> annotations = 4;
+}
+
+// PodSandboxStats provides the resource usage statistics for a pod.
+// The linux or windows field will be populated depending on the platform.
+message PodSandboxStats {
+    // Information of the pod.
+    PodSandboxAttributes attributes = 1;
+    // Stats from linux.
+    LinuxPodSandboxStats linux = 2;
+    // Stats from windows.
+    WindowsPodSandboxStats windows = 3;
+}
+
+// LinuxPodSandboxStats provides the resource usage statistics for a pod on linux
+message LinuxPodSandboxStats {
+    // CPU usage gathered from the pod.
+    CpuUsage cpu = 1;
+    // Memory usage gathered from the pod.
+    MemoryUsage memory = 2;
+    // Stats pertaining to CPU resources consumed by pod cgroup (which includes all containers' resource usage and pod overhead).
+    NetworkUsage network = 3;
+    // Stats pertaining to processes in the pod.
+    ProcessUsage process = 4;
+    // Stats of containers in the measured pod.
+    repeated ContainerStats containers = 5;
+}
+
+// WindowsPodSandboxStats provides the resource usage statistics for a pod on windows
+message WindowsPodSandboxStats {
+    // TODO: Add stats relevant to windows
+}
+
+// NetworkUsage contains data about network resources.
+message NetworkUsage {
+    // The time at which these stats were updated.
+    int64 timestamp = 1;
+    // Stats for the default interface, if found
+    NetworkInterfaceUsage default_interface = 2;
+    // Stats for all found interfaces, excluding the default
+    repeated NetworkInterfaceUsage interfaces = 3;
+}
+
+// NetworkInterfaceUsage contains resource value data about a network interface.
+message NetworkInterfaceUsage {
+    // The name of the interface
+    string name = 1;
+    // Cumulative count of bytes received.
+    UInt64Value rx_bytes = 2;
+    // Cumulative count of receive errors encountered.
+    UInt64Value rx_errors = 3;
+    // Cumulative count of bytes transmitted.
+    UInt64Value tx_bytes = 4;
+    // Cumulative count of transmit errors encountered.
+    UInt64Value tx_errors = 5;
+}
+
+// ProcessUsage are stats pertaining to processes.
+message ProcessUsage {
+    // The time at which these stats were updated.
+    int64 timestamp = 1;
+    // Number of processes in the pod.
```

Comment:
maybe delete "in the pod" since the message could be potentially reused for containers stats too

Reply:
good point